addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/core/LxControl.java

```diff
@@ -0,0 +1,297 @@
+     * Update Miniserver's control in runtime.
+     *
+     * @param json
+     *            JSON describing the control as received from the Miniserver
+     * @param room
+     *            New room that this control belongs to
+     * @param category
+     *            New category that this control belongs to
+     */
+    void update(LxJsonControl json, LxContainer room, LxCategory category) {
+
+        logger.trace("Updating LxControl: {}", json.type);
+
+        this.name = json.name;
+        this.room = room;
+        this.category = category;
+        uuid.setUpdate(true);
+        if (room != null) {
+            room.addOrUpdateControl(this);
+        }
+        if (category != null) {
+            category.addOrUpdateControl(this);
+        }
+
+        // retrieve all states from the configuration
+        if (json.states != null) {
+
+            logger.trace("Reading states for LxControl: {}", json.type);
+
+            for (Map.Entry<String, JsonElement> jsonState : json.states.entrySet()) {
+                JsonElement element = jsonState.getValue();
+                if (element instanceof JsonArray) {
+                    // temperature state of intelligent home controller object is the only
+                    // one that has state represented as an array, as this is not implemented
+                    // yet, we will skip this state
+                    continue;
+                }
+                String value = element.getAsString();
+                if (value != null) {
+                    LxUuid id = new LxUuid(value);
+                    String name = jsonState.getKey().toLowerCase();
+                    LxControlState state = states.get(name);
+                    if (state == null) {
+                        logger.trace("New state for LxControl {}: {}", json.type, name);
+                        state = new LxControlState(id, name, this);
+                    } else {
+                        logger.trace("Existing state for LxControl {}: {}", json.type, name);
+                        state.getUuid().setUpdate(true);
+                        state.setName(name);
+                    }
+                    states.put(name, state);
+                }
+            }
+        }
+    }
+
+    /**
+     *
+     * @param client
+     *            websocket client to facilitate communication with Miniserver
+     * @param uuid
+     *            UUID of the control to be created
+     * @param json
+     *            JSON describing the control as received from the Miniserver
+     * @param room
+     *            Room that this control belongs to
+     * @param category
+     *            Category that this control belongs to
+     * @return
+     *         created control object or null if error
+     */
+    static LxControl createControl(LxWsClient client, LxUuid uuid, LxJsonControl json, LxContainer room,
+            LxCategory category) {
+
+        if (json == null || json.type == null || json.name == null) {
```

Comment:
I am in the process of moving the whole code to non-null by default. Since this is a significant change, I'd like to request a separate review and create a new PR. For this PR, could we keep it the way it is now, with these possibly redundant checks?